drivers/flash/Kconfig.mcux

```diff
@@ -27,6 +27,18 @@ endif # SOC_FLASH_MCUX
 
 if HAS_MCUX_FLEXSPI
 
+config USE_DEDICATED_FLEXSPI_FLASH_DRIVER
+	bool "Use a dedicated flexspi flash driver"
+	help
+	  Enables choosing a flexspi flash driver.
+
+if USE_DEDICATED_FLEXSPI_FLASH_DRIVER
+
+choice FLEXSPI_FLASH_DRIVER
+	prompt "Flexspi flash driver"
+	help
+	  Select the flexspi driver matching your device.
+
```

Comment:
you don't need a double layer here:
![image](https://user-images.githubusercontent.com/9802583/134484417-ad17228e-b26e-46e5-8df3-952d36e54aac.png)

If using the `optional` flag, you'll get this instead:
![image](https://user-images.githubusercontent.com/9802583/134484951-d1fc0e54-2533-4c79-9a23-550511005e21.png)

Reply:
This was changed to not use double layer.